src/peerbook/libp2p_peerbook.erl

```diff
@@ -562,3 +563,27 @@ install_gossip_handler(TID, Handle) ->
             libp2p_group_gossip:add_handler(G,  ?GOSSIP_GROUP_KEY, {?MODULE, Handle}),
             G
     end.
+
+get_signed_metadata(State = #state{tid=TID, metadata_ref=MR}) ->
+    case MR of
+        undefined ->
+            MetaDataFun = libp2p_config:get_opt(libp2p_swarm:opts(TID), [libp2p_peerbook, signed_metadata_fun],
```

Comment:
Do we have to look up metadata_fun every time? Why not just get it on peer book init and pass store it in state?

Reply:
it;s an ETS lookup, it should be fast?

Reply:
but there's no need to?